src/Compilers/Core/AnalyzerDriver/AnalyzerExecutor.cs

```diff
@@ -25,6 +26,8 @@ internal class AnalyzerExecutor
         private readonly Action<Exception, DiagnosticAnalyzer, Diagnostic> _onAnalyzerException;
         private readonly AnalyzerManager _analyzerManager;
         private readonly Func<DiagnosticAnalyzer, bool> _isCompilerAnalyzer;
+        private readonly ImmutableDictionary<DiagnosticAnalyzer, object> _singleThreadedAnalyzerToGateMap;
```

Comment:
`CompilationWithAnalyzers` is not a subtype of `Compilation`, it is a class that contains a `Compilation` as a field, so even that cannot happen.